common/message_algos/src/filters/dedup_filter.rs

```diff
@@ -0,0 +1,54 @@
+//! Message deduplication
+
+use crate::{filters::*, Envelope};
+use std::collections::VecDeque;
+
+/// Test whether two messages are a duplicates.
+pub trait DedupPolicy: Send {
+    type Message: Send + Clone;
+
+    fn is_duplicate(&self, msg1: &Envelope<Self::Message>, msg2: &Envelope<Self::Message>) -> bool;
+}
```

Comment:
I see as not appropriate for dedup.

Somehow one needs an aggregate of the previous messages and checking for duplicate has to be done against that aggregate.

Reply:
yes, and this is done. See `DedupeFilter` implementation below. Think of `DedupPolicy` as `Eq`.